jhu/delphi_jhu/geo.py

```diff
@@ -30,22 +31,29 @@ def geo_map(df: pd.DataFrame, geo_res: str):
     df = df.copy()
     if geo_res not in VALID_GEO_RES:
         raise ValueError(f"geo_res must be one of {VALID_GEO_RES}")
-
+    unassigned_counties = df[df["fips"].str.endswith("000")].copy()
+    df = df[df["fips"].astype(int) % 1000 != 0].copy()
     gmpr = GeoMapper()
     if geo_res == "county":
-        df.rename(columns={'fips': 'geo_id'}, inplace=True)
-    elif geo_res == "state":
-        df = df.set_index("fips")
-        # Zero out the state FIPS population to avoid double counting.
-        state_fips_codes = {str(x).zfill(2) + "000" for x in range(1,73)}
-        subset_state_fips_codes = set(df.index.values) & state_fips_codes
-        df.loc[subset_state_fips_codes, "population"] = 0
-        df = df.reset_index()
-        df = gmpr.replace_geocode(df, "fips", "state_id", new_col="geo_id", date_col="timestamp")
+        if not sensor.endswith("_prop"):
+            # It is not clear how to calculate the proportion for unallocated
+            # cases/deaths, so we exclude them for those sensors.
+            df = df.append(unassigned_counties)
+        df.rename(columns={"fips": "geo_id"}, inplace=True)
+    elif geo_res in ("state", "hhs", "nation"):
+        state_geo = "state_id" if geo_res == "state" else "state_code"
+        df = df.append(unassigned_counties)
+        df = gmpr.replace_geocode(df, "fips", state_geo, new_col="geo_id", date_col="timestamp")
+        df.drop("population", axis=1, errors="ignore", inplace=True)
+        df = gmpr.add_population_column(df, state_geo, geocode_col="geo_id")
+        df.drop("state_id", axis=1, errors="ignore", inplace=True)
```

Comment:
The tests in test_geo.py contain props, though if there's a simpler way to achieve this logic i'm all for it. still need to read into your comment properly

Reply:
ok I see what you mean. Definitely having the add_population in pull.py with dropna=True causes some problems. I'll move that statement to here so we don't have 2 add population statements in different files. The reason line 48 exists is because we dont want to aggregate the sum of counties for the state counts, we just want to use the state population. Otherwise, we'll underestimate the total population because the megafips counties got thrown away. (Unless the "unallocated" is independent of how many standard fips they report, and they always report all the fips).

Reply:
The cases/deaths sources always report all fips. There is no unaccounted-for population to put in the megafips for cases/deaths.

Reply:
From the doc:
```
JHU/USAFacts
Current 
    Numerator
        -Sum of all county + unassigned counts
    Denominator
        -sum of all reported county populations (99% of the time should be = to state population) (assumes unassigned counts are distributed only among reported counties)
Proposed
    Numerator
        -Sum of all county + unassigned counts
    Denominator
        -State population
```
My intention was to implement the "State population" as the denominator, but are you saying we're now ok with the "sum of all reported county populations" now since they should be identical (or close enough)?

Reply:
I think we trailed off a [Slack conversation](https://delphi-org.slack.com/archives/G01E20QCREH/p1612564916137400) to that effect, yeah

Reply:
:+1: will do